File: Louis991-cs50-problems-2020-x-cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    //Setting global variables: amount owed and count, the min. number of coins.
    float owed;
    int count = 0;
    
    //Get user input, in $.
    do
    {
        owed = get_float("Enter the amount owed: ");
    }
    while (owed <= 0);
    
    //Convert amount owed to cents, which implies changing the data type of the input from float to int.
    int cents = round(owed * 100);
    
    //Calculate the minimum amount of change necessary.
    if (cents >= 25)
    {
        do
        {
            cents -= 25;
            count += 1;
        }
        while (cents >= 25);
    }
    
    if (cents >= 10)
    {
        do
        {
            cents -= 10;
            count += 1;
        }
        while (cents >= 10);
    }
    
    if (cents >= 5)
    {
        do
        {
            cents -= 5;
            count += 1;
        }
        while (cents >= 5);
    }
    
    if (cents >= 1)
    {
        do
        {
            cents -= 1;
            count += 1;
        }
        while (cents >= 1);
    }
    //Print the min. number of coins required.
    printf("%i\n", count);
}

File: Louis991-cs50-problems-2020-x-sentimental-cash/cash.py
from cs50 import get_float

# Amount owed and counter for the minimum number of coins required.
owed = 0
count = 0

# Getting user input.
while owed <= 0:
    owed = get_float("Enter the amount owed: ")

# Converting the amount to cents
owed = round(owed * 100)

# Determining the min. amount of coins
if owed >= 25:
    while owed >= 25:
        owed -= 25
        count += 1

if owed >= 10:
    while owed >= 10:
        owed -= 10
        count += 1

if owed >= 5:
    while (owed >= 5):
        owed -= 5
        count += 1

if owed >= 1:
    while (owed >= 1):
        owed -= 1
        count += 1

# Printing the output
print(f"{count}")